lib/modules.nix

```diff
@@ -554,11 +554,11 @@ let
               else
                 mapAttrs
                   (n: option:
-                    [{ inherit (module) _file; options = option; }]
+                    [{ inherit (module) _file; pos = builtins.unsafeGetAttrPos n subtree; options = option; }]
```

Comment:
Should we fall back to the position of the parent, somewhat similar to what we do with `_file` in `imports`? (where the `imports` relation corresponds to the option nesting relation, and the partiality comes from anonymous modules or generated `options` attrs, resp.)

Reply:
Not necessarily, since we get garbage for a lot of options anyway, so this wouldn't improve that much I don't think.

```
dev/nixpkgs2 » rg attrsets.nix ../nixpkgs/opts-tags | wc -l
957
dev/nixpkgs2 » cat ../nixpkgs/opts-tags | wc -l
11201
```